src/carbon/adapters/smtp_adapter.cr

```diff
@@ -64,4 +64,12 @@ class Carbon::SmtpAdapter < Carbon::Adapter
       {username, password}
     end
   end
+
+  private def get_helo_domain : String
+    if settings.helo_domain.nil?
+      raise "'helo_domain' setting is not provided"
+    end
+
+    return settings.helo_domain
+  end
```

Comment:
Is this actually necessary? I think you removed nilability from `helo_domain` above so it should not ever be `nil`, or am I missing something?